src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets

```
@@ -343,10 +343,23 @@ Copyright (c) .NET Foundation. All rights reserved.
     
   </Target>
 
-  <Target Name="CheckWindowsDesktopSdkInUse" 
-          BeforeTargets="_CheckForInvalidConfigurationAndPlatform" 
-          Condition="'$(UseWpf)' == 'true' Or '$(UseWindowsForms)' == 'true'">
-    <NetSdkWarning Condition="'$(_MicrosoftNetSdkWindowsDesktop)' != 'true'" 
-                   ResourceName="UseWpfOrUseWindowsFormsRequiresWindowsDesktopFramework" />
+  <Target Name="_CheckForTransitiveWindowsDesktopDependencies"
+          AfterTargets="ResolvePackageAssets"
+          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0')) and '$(TargetPlatformIdentifier)' != 'Windows' and '@(TransitiveFrameworkReference)' != ''">
+    <NetSdkError Condition="@(TransitiveFrameworkReference->'%(Identity)'->Contains('Microsoft.WindowsDesktop.App'))"
+                 ResourceName="WindowsDesktopTargetPlatformMustBeWindows" />
+  </Target>
+
+  <Target Name="_CheckForInvalidWindowsDesktopTargetingConfiguration"
+        BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
+        Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0')) and ('$(TargetPlatformIdentifier)' == 'Windows' or '$(UseWindowsForms)' == 'true' or '$(UseWPF)' == 'true')">
```

I think you can drop the check that the TargetPlatformIdentifier is Windows, because it's actually going to be generating an error if that's not the case.

```suggestion
        Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0')) and ('$(UseWindowsForms)' == 'true' or '$(UseWPF)' == 'true')">
```